platform/default/src/mbgl/util/thread_local.cpp

```diff
@@ -11,56 +8,37 @@
 
 namespace mbgl {
 namespace util {
-
-template <class T>
-class ThreadLocal<T>::Impl {
-public:
-    pthread_key_t key;
-};
-
-template <class T>
-ThreadLocal<T>::ThreadLocal() : impl(std::make_unique<Impl>()) {
-    int ret = pthread_key_create(&impl->key, [](void *) {});
-
-    if (ret) {
-        throw std::runtime_error("Failed to init local storage key.");
-    }
+namespace impl {
+
+ThreadLocalBase::ThreadLocalBase() {
+    static_assert(sizeof(storage) >= sizeof(pthread_key_t), "storage is too small");
+    static_assert(alignof(decltype(storage)) % alignof(pthread_key_t) == 0, "storage is incorrectly aligned");
+    if (const int ret = pthread_key_create(&reinterpret_cast<pthread_key_t&>(storage), nullptr)) {
+        throw std::runtime_error("Failed to init local storage key");
```

Comment:
Would be nice to avoid exceptions, how do we handle this one?

Reply:
maybe we could just `Log::Error.. ` or even abort instead?

Reply:
I don't think these exceptions are handled, at least, I could not find any place where they would.

Reply:
I was just porting the code (we had exceptions before too), but I've never seen them fail, and since they're unhandled anyway, it probably makes sense to abort the program here? After all, we're using them to store the Scheduler/RunLoop, and if we can't do that then there's probably no point in continuing to execute the program at all.